src/sentry/static/sentry/app/components/events/interfaces/debugMeta/debugFileDetails/index.tsx

```diff
@@ -87,18 +87,43 @@ class DebugFileDetails extends AsyncComponent<Props, State> {
       return candidates;
     }
 
-    return candidates.map(candidate => {
-      if (!debugFiles.find(debugFile => debugFile.id === candidate.location)) {
+    // Check for unapplied debug files
+    const unAppliedDebugFiles: Candidates = [];
+
+    for (const debugFileIndex in debugFiles) {
+      const debugFile = debugFiles[debugFileIndex];
+
+      if (candidates.find(candidate => candidate.location === debugFile.id)) {
+        continue;
+      }
+
+      unAppliedDebugFiles.push({
+        download: {
+          status: CandidateDownloadStatus.UNAPPLIED,
+        },
+        location: debugFile.id,
+        source: INTERNAL_SOURCE,
+        source_name: debugFile.objectName,
+      });
+    }
```

Comment:
Should we be mixing functional and imperative styles here? This might read better if it was a `filter() + map()`